ysh/expr_eval.py

```diff
@@ -687,10 +687,18 @@ def _Concat(self, left, right, op):
             c.extend(right.items)
             return value.List(c)
 
+        elif left.tag() == value_e.Dict and right.tag() == value_e.Dict:
+            left = cast(value.Dict, UP_left)
+            right = cast(value.Dict, UP_right)
+
+            res = left.d.copy()
```

Comment:
Hm technically if we wanted to keep the mycpp runtime small, and reduce total code

We could just inline the copy, and avoid the copy() method:

```
res = {}
for ...

```

It is more efficient to do the reserve() though, so I guess it's fine to keep it.  One more thing for a Rust translation :-)

---

Actually now I realized the C++ version can be optimized even more by copying the 4 structs/slabs (Dict, index, keys, values)

which avoids hashing

So I guess we can keep the copy() method  and maybe do it later